jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/web/WebStepsUtils.java

```diff
@@ -0,0 +1,53 @@
+package org.jbehavesupport.core.internal.web;
+
+import lombok.RequiredArgsConstructor;
+import org.jbehavesupport.core.web.WebAction;
+import org.jbehavesupport.core.web.WebActionContext;
+import org.jbehavesupport.core.web.WebActionResolver;
+import org.jbehavesupport.core.web.WebProperty;
+import org.jbehavesupport.core.web.WebPropertyContext;
+import org.jbehavesupport.core.web.WebPropertyResolver;
+import org.jbehavesupport.core.web.WebWaitCondition;
+import org.jbehavesupport.core.web.WebWaitConditionContext;
+import org.jbehavesupport.core.web.WebWaitConditionResolver;
+import org.openqa.selenium.StaleElementReferenceException;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+
+@RequiredArgsConstructor
+@Component
+public class WebStepsUtils {
+    public static final String ACTION = "action";
+    public static final String ELEMENT = "element";
+    public static final String PROPERTY = "property";
+
+    private final WebPropertyResolver propertyResolver;
+    private final WebWaitConditionResolver waitConditionResolver;
+    private final WebActionResolver actionResolver;
+
+    @Retryable(value = { StaleElementReferenceException.class })
+    public Object resolvePropertyValue(String page, Map<String, String> row) {
+        WebProperty property = propertyResolver.resolveProperty(row.get(PROPERTY));
+
+        WebPropertyContext ctx = WebPropertyContext.builder()
+            .page(page)
+            .element(row.get(ELEMENT))
+            .build();
+
+        return property.value(ctx);
+    }
+
+    @Retryable(value = { StaleElementReferenceException.class })
+    public void evaluateWebWaitCondition(WebWaitConditionContext waitConditionCtx){
+        WebWaitCondition waitCondition = waitConditionResolver.resolveWaitCondition(waitConditionCtx);
+        waitCondition.evaluate(waitConditionCtx);
+    }
+
+    @Retryable(value = { StaleElementReferenceException.class })
+    public void performAction(WebActionContext actionContext, String actionName){
+        WebAction action = actionResolver.resolveAction(actionName);
+        action.perform(actionContext);
+    }
+}
```

Comment:
I don't think that introducing "internal util class" is the right approach. What if somebody is using WebAction/WebProperty directly, wouldn't this be beneficial for him too? I'd say that if we add this feature it should work OOTB.

Reply:
We can make it public.